packages/net/tcp_connection.pony

```diff
@@ -657,7 +648,7 @@ actor TCPConnection
       var bytes_to_send: USize = 0
       var bytes_sent: USize = 0
       while _writeable and (_pending_writev_total > 0) do
-        if bytes_sent >= _max_size then
+        if bytes_sent >= _read_buffer_size then
```

Comment:
I opened an issue #3186 to address this issue. I think it should be a different PR.